hadoop-hdds/docs/content/design/event-notifications.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Event notification support in Ozone
+summary: Event notifications for all bucket/event types in ozone
+date: 2025-06-28
+jira: HDDS-13513
+status: design
+author: Donal Magennis, Colm Dougan
+---
+<!--
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+
+# Abstract
+
+Implement an event notification system for Apache Ozone, providing the ability for users to consume events occuring on the Ozone filesystem.
+This is similar to https://issues.apache.org/jira/browse/HDDS-5984 but aims to encapsulate all events and not solely S3 buckets.  
+This document proposes a potential solution and discusses some of the challenges/open questions.
+
+# Introduction
+
+Apache Ozone does not currently provide the ability to consume filesystem events, similar to how HDFS does with Inotify or S3 with bucket notifications.  
+These events are an integral part of integration with external systems to support real-time, scalable, and programmatic monitoring of changes in the data or metadata stored in Ozone.  
+These external systems can use notifications of objects created/deleted to trigger data processing workflows, replication and monitoring alerts.
+
+# Goals
+
+Provide support for all events across the Ozone filesystem for FSO and non FSO buckets, including renames and changes to acls.
+Not impact with performance of client requests.
+Guarantee at-least-once delivery.
+
+# Non-Goals
+
+Filtering of events or paths/buckets
+Persistent storage of notification messages
+Asynchronous delivery
```

Comment:
High availability and consistency.

It sounds like the design does not consider availability. If the Listener goes down there's no guarantee the events notification will be processed.

Consistency: does it guarantee in-order delivery of events? Any delivery latency assumptions or SLAs?